### ISSUE TITLE
linter: exhaustive-deps now requires full object to be defined in the dependency array

### ISSUE THREAD

**Issue by MattisAbrahamsson** [oxc-project/oxc#15796]
### What version of Oxlint are you using?

1.29.0

### What command did you run?

`oxlint --quiet --config=.oxlintrc.json --ignore-path=.gitignore`

### What happened?

After upgrading to 1.29.0 I received `exhaustive-deps` warnings that were not present in previous versions, it happens when the dependency array defines an object property like this:
```ts
useMemo(() => {
console.log(obj.myValue)
}, [obj.myValue]);
```
Oxlint wants the entire `obj` to be defined in the dep array.

This regression is likely related to [this change](https://github.com/oxc-project/oxc/pull/13902)

**Comment by camc314** [comment]
Thank you for reporting this issue! I've investigated the changes in PR #13902 and the behavior of the exhaustive-deps rule.

**Summary of my findings:**

PR #13902 fixed a bug where member expressions used as arguments (e.g., `obj.myValue` in `console.log(obj.myValue)`) were incorrectly treated as if they were callees. The old implementation would track only the base object (`obj`) instead of the full property chain (`obj.myValue`).

**Expected behavior (current):**
```tsx
function MyComponent() {
  const obj = { myValue: 42 };
  useMemo(() => {
    console.log(obj.myValue); // Accessing the property
  }, [obj.myValue]); // ✅ Correct - should not warn
}
```

**Could you please provide:**
1. A minimal reproducible code example that shows the warning you're seeing
2. The exact error message from oxlint
3. Whether `obj.myValue` is being called as a function (e.g., `obj.myValue()`) or just accessed

The current implementation correctly handles property access vs. method calls:
- `obj.myValue` (property access) → requires `[obj.myValue]`  
- `obj.myValue()` (method call) → requires `[obj]`

I've added test cases to ensure this behavior is preserved. If you can provide more details about your specific scenario, I can investigate further.

**Comment by camc314** [comment]
Can you please share more infomation? Or ideally a git repo with a reproduction of the issue?  I can't reproduce this issue:

<img width="953" height="320" alt="Image" src="https://github.com/user-attachments/assets/877e0842-c4b7-4c25-9c5b-de5336de0fc7" />

**Comment by connorshea** [comment]
I managed to reproduce this: https://github.com/connorshea/repro-oxlint-react-issue

`git clone https://github.com/connorshea/repro-oxlint-react-issue && cd repro-oxlint-react-issue && npm install && npm run oxlint` to see the issue.

It's a bit arbitrary but I can confirm it is a new violation vs 1.28:
```tsx
import { useMemo } from 'react';

interface Props {
  file: File;
  showSize?: boolean;
}

export const FileSize = ({ file, showSize = true }: Props) => {
  const fileSizeInMB = useMemo(
    () => (showSize ? (file.size / (1024 * 1024)).toFixed(2) : undefined),
    [showSize, file.size], // Fails on oxlint v1.29, does not fail on eslint or oxlint v1.28
  );

  return <p>{fileSizeInMB} MB</p>;
};
```

**Comment by camc314** [comment]
thanks connor, i'll take a look tomorrow

**Comment by MattisAbrahamsson** [comment]
> I managed to reproduce this: https://github.com/connorshea/repro-oxlint-react-issue
> 
> `git clone https://github.com/connorshea/repro-oxlint-react-issue && cd repro-oxlint-react-issue && npm install && npm run oxlint` to see the issue.
> 
> It's a bit arbitrary but I can confirm it is a new violation vs 1.28:
> 
> import { useMemo } from 'react';
> 
> interface Props {
>   file: File;
>   showSize?: boolean;
> }
> 
> export const FileSize = ({ file, showSize = true }: Props) => {
>   const fileSizeInMB = useMemo(
>     () => (showSize ? (file.size / (1024 * 1024)).toFixed(2) : undefined),
>     [showSize, file.size], // Fails on oxlint v1.29, does not fail on eslint or oxlint v1.28
>   );
> 
>   return <p>{fileSizeInMB} MB</p>;
> };

Thanks for making a reproduction!

**Comment by camc314** [comment]
fixed in #15832
